manifests/init.pp

```diff
@@ -0,0 +1,130 @@
+# @summary Manage a MySQL/MariaDB server configuration.
+#
+# @param create_mysql_home
+# Should the mysql_home be created?
+#
+# @param dbs
+# Raw data to define mysql::db resources.
+#
+# @param mysql_gid
+# The GID for the mysql user.
+#
+# @param mysql_groupname
+# The name of the mysql group.
+#
+# @param mysql_home
+# The home dir of the mysql user.
+#
+# @param mysql_logdir
+# Location of MySQL log files.
+#
+# @param mysql_parentdir
+# Location of directory that contains the mysql_home.
+# NOTE: Might be useful for this to be an array so multiple nested
+# dirs can be created as dependencies.
+#
+# @param mysql_rundir
+# Location of run directory (generally a sub-directory of /var/run/).
+#
+# @param mysql_uid
+# UID of the mysql user.
+#
+# @param mysql_username
+# Username of the mysql user.
+#
+# @param other_dependencies
+# Other depencies to handle prior to other MySQL setup, specified as resources, e.g.:
+#   - "Mount['/var/lib/mysql']"
+#
+# @param yumrepo
+# Raw params for a yumrepo resource from which to install MySQL/MariaDB.
+#
+# @example
+# include profile_mysql_server
+#
+class profile_mysql_server (
+
+  Boolean           $create_mysql_home,
+  Hash              $dbs,
+  Integer           $mysql_gid,
+  String            $mysql_groupname,
+  String            $mysql_home,
+  String            $mysql_logdir,
+  Optional[String]  $mysql_parentdir,
```

Comment:
Optional parameters should be listed after required parameters.

Reply:
Fixed that one.